apps/envited-x-data-space/common/validator/shacl/shacl.utils.ts

```diff
@@ -15,6 +16,20 @@ export const validateShacl = (shapes: DatasetCore<Quad, Quad>) => async (data: D
   return validator.validate(data)
 }
 
+export const checkIfAssetExists = async (file: File) => {
+  const arrayBuffer = Buffer.from(await file.arrayBuffer())
+  const cid = await createFilename(arrayBuffer)
+  const response = await fetch('/api/check-asset-exists', {
```

Comment:
For http calls on the UI side it's a good idea to use something like useSWR, it allows you to easily handle loading states, retries and streaming. You can use fetch inside useSWR as the tool that actually makes the call.

Reply:
I tried to use SWR for this but I need the CID for the request and I'm not able to calculate the CID without async. This will fail the UploadAssetItem.tsx element. I'm also able to use SWR in an useEffect as the current validation is working